### PR TITLE
feat: Python Setup integration

### PR DESCRIPTION
**Pull request by vibhatha**
This PR includes a modification to the existing setup and refactoring of existing helper classes and test cases. 

The changes reflects the following

- [x] Adding setup.py
- [x] Organized utils and core classes and tests to be packaged
- [x] Setup requirements clarified and documented

### CODE REVIEW

**Review by westonpace — COMMENTED** [review]
These changes appear to be mostly cosmetic.  I'll defer to whatever @richtia thinks is best.  @richtia , does this look correct?

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        'pyarrow',
+        'pytest',
+        'pytest-xdist',
+        'substrait-validator',
+    ],
+    dependency_links=[
+      '--extra-index-url https://pypi.fury.io/arrow-nightlies --prefer-binary --pre'
+    ],
+    package_data={
+        'tests': ['tests/integration'],
```
Comment (richtia) [inline setup.py:27]: should tests/functional be here too?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from setuptools import setup, find_packages
+setup(
+    name='substrait-consumer',
```
Comment (richtia) [inline setup.py:3]: I think this project has grown to a point where it's testing both producers and consumers.  maybe just substrait-tests? and change the folder to substrait_tests as well?  we can get feedback from others too.

File: setup.py (hunk 1 above)
Comment (vibhatha) [reply setup.py:27]: Ah yes, good catch.

File: setup.py (hunk 2 above)
Comment (vibhatha) [reply setup.py:3]: Yeah, probably we need to update the repo name too. Let’s decide a better name. +1

Comment (vibhatha) [reply setup.py:3]: cc @jacques-n @westonpace

**Comment by vibhatha** [comment]
@richtia I updated the README as well.

**Comment by vibhatha** [comment]
Also, immediately after this PR we could also create a github workflow to at least test the core features of the package. WDYT @richtia?

**Comment by richtia** [comment]
> Also, immediately after this PR we could also create a github workflow to at least test the core features of the package. WDYT @richtia?

I think that would be a great idea!

**Comment by richtia** [comment]
@vibhatha did you run into errors trying to do the install?
```error: The 'sqlalchemy<2,>=1' distribution was not found and is required by ibis-substrait```

**Comment by vibhatha** [comment]
Closing this in favor of https://github.com/substrait-io/consumer-testing/pull/12

**Pull request closed without merging**